crates/kornia-apriltag/src/lib.rs

```diff
@@ -27,3 +45,455 @@ pub mod quad;
+    pub fit_quad_config: FitQuadConfig,
+    /// Whether to enable edge refinement before decoding.
+    pub refine_edges_enabled: bool,
+    /// Sharpening factor applied during decoding.
+    pub decode_sharpening: f32,
+    /// Whether normal border tags are present.
+    pub normal_border: bool,
+    /// Whether reversed border tags are present.
+    pub reversed_border: bool,
+    /// Minimum tag width at border among all families.
+    pub min_tag_width: usize,
+    /// Minimum difference between white and black pixels for thresholding.
+    pub min_white_black_difference: u8,
+}
+
+impl DecodeTagsConfig {
+    /// Creates a new `DecodeTagsConfig` with the given tag families.
+    pub fn new(tag_families: Vec<TagFamily>) -> Self {
+        let mut normal_border = false;
+        let mut reversed_border = false;
+        let mut min_tag_width = usize::MAX;
+
+        tag_families.iter().for_each(|family| {
+            if family.width_at_border < min_tag_width {
+                min_tag_width = family.width_at_border;
+            }
+            normal_border |= !family.reversed_border;
+            reversed_border |= family.reversed_border;
+        });
+
+        if min_tag_width < 3 {
+            min_tag_width = 3;
+        }
+
+        // TODO: Set the default values which are best for the given tags
+        Self {
+            tag_families,
+            fit_quad_config: Default::default(),
+            normal_border,
+            refine_edges_enabled: true,
+            decode_sharpening: 0.15,
+            reversed_border,
+            min_tag_width,
+            min_white_black_difference: 20,
+        }
+    }
+
+    /// Creates a `DecodeTagsConfig` with all supported tag families.
+    pub fn all() -> Self {
+        let tag_families = vec![
+            TagFamily::tag16_h5(),
+            TagFamily::tag25_h9(),
+            TagFamily::tag36_h10(),
+            TagFamily::tag36_h11(),
+            TagFamily::tagcircle21_h7(),
+            TagFamily::tagcircle49_h12(),
+            TagFamily::tagstandard41_h12(),
+            TagFamily::tagstandard52_h13(),
+            TagFamily::tagcustom48_h12(),
+        ];
+
+        Self::new(tag_families)
+    }
+
+    /// Adds a tag family to the configuration.
+    pub fn add(&mut self, family: TagFamily) {
+        if family.width_at_border < self.min_tag_width {
+            self.min_tag_width = family.width_at_border;
+        }
+        self.normal_border |= !family.reversed_border;
+        self.reversed_border |= family.reversed_border;
+
+        self.tag_families.push(family);
+    }
+
+    /// Resets the sharpening buffer for each tag family in the configuration.
+    pub fn reset(&mut self) {
+        self.tag_families.iter_mut().for_each(|family| {
+            family.sharpening_buffer.reset();
+        });
+    }
+}
+
+/// Decoder for AprilTag detection and decoding.
+pub struct AprilTagDecoder {
+    /// Configuration for decoding AprilTags.
+    pub config: DecodeTagsConfig,
```

Comment:
keep all private and implement a pub inline getter for consistency with the rest of the struct